PracticeGit/intlist.c

```diff
@@ -0,0 +1,152 @@
+//----------------------------------------------------------------------------------------------
+// Training ~ A training program for new joiners at Metamation, Batch - July 2024.
+// Copyright (c) Metamation India.
+//------------------------------------------------------------------------------------------------
+//   intlist.c
+//   Program on Linked List functions.
+//--------------------------------------------------------------------------------------------
+#include <malloc.h>
+#include "intlist.h"
+
+// Initialize the head pointer of the linked list to NULL.
+struct node* head = NULL;
+
+// Function to create a empty linked list
+struct node* CreateList (int initialData) {
+	struct node* head = (struct node*)malloc (sizeof (struct node));
+	if (head == NULL) {
+		return ERROR_MEM_ALLOC;
+	}
+	head->data = initialData;
+	head->next = NULL;
+	return head;
+}
+
+//	Function to add an element to the end of the list
+void Append (struct node* head, int newData) {
+	struct node* current = head;
+	while (current->next != NULL) {
+		current = current->next;
+	}
+	struct node* newNode = (struct node*)malloc (sizeof (struct node));
+	if (newNode == NULL) {
+		return ERROR_MEM_ALLOC;
+	}
+	newNode->data = newData;
+	newNode->next = NULL;
+	if (head == NULL) {
+		// If list is empty, make newNode the head

```

Comment:
It is recommended not to put {} around single line blocks. It reduces readability.